libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SoqlSyncDownTarget.java

```diff
@@ -130,6 +138,80 @@ public JSONArray continueFetch(SyncManager syncManager) throws IOException, JSON
         return records;
     }
 
+    @Override
+    public Set<String> getListOfRemoteIds(SyncManager syncManager, Set<String> localIds) {
```

Comment:
The paging bit is already there in startFetch/continueFetch.

Maybe you should create another SOQLSyncDownTarget with the modified soql query and call startFetch/continueFetch on it to avoid duplicating code. In that case, instead of having a method to get remote ids, you would have a method on SOQLSyncDownTarget to get back the ids-only SyncDownTarget?

Or alternatively, you could refactor startFetch/continueFetch to have an extra parameter: queryToRun. You would have 

```java
startFetch(SyncManager syncManager, long maxTimeStamp) {
  queryToRun = ...
  startFetch(syncManager, maxTimeStamp, queryToRun)
}

startFetch(SyncManager syncManager, long maxTimeStamp, queryToRun) {
  // pretty much the current implementation
}
```
In that approach, the getRemoteIds would call startFetch with the ids-only query.

Personally, I kind of like the first approach better, the looping through startFetch/continueFetch would be done in the sync manager.